File: code_challenge_solution/c#/169 Rock Paper Scissors.cs
using System;

namespace RockPaperScissors
{
    class Program
    {
        static void Main(string[] args)
        {
            string[] choices = { "rock", "paper", "scissors" };

            Console.WriteLine("Let's play Rock Paper Scissors!");

            while (true)
            {
                // Get user's choice
                Console.WriteLine("Enter your choice (rock, paper, or scissors): ");
                string userChoice = Console.ReadLine().ToLower();

                // Generate computer's choice
                Random random = new Random();
                int index = random.Next(choices.Length);
                string computerChoice = choices[index];

                // Determine the winner
                Console.WriteLine("The computer chose: " + computerChoice);

                if (userChoice == computerChoice)
                {
                    Console.WriteLine("It's a tie!");
                }
                else if (userChoice == "rock" && computerChoice == "scissors" || 
                         userChoice == "paper" && computerChoice == "rock" || 
                         userChoice == "scissors" && computerChoice == "paper")
                {
                    Console.WriteLine("You win!");
                }
                else
                {
                    Console.WriteLine("Computer wins!");
                }

                Console.WriteLine("");
                Console.WriteLine("Do you want to play again? (yes or no)");
                string playAgain = Console.ReadLine().ToLower();

                if (playAgain != "yes")
                {
                    break;
                }
                Console.Clear();
            }
        }
    }
}

File: code_challenge_solution/python/169 Rock Paper Scissors.py
import random

def get_user_choice():
    user_choice = input("Enter your choice (rock, paper, or scissors): ")
    user_choice = user_choice.lower()
    return user_choice

def get_computer_choice():
    choices = ["rock", "paper", "scissors"]
    computer_choice = random.choice(choices)
    return computer_choice

def determine_winner(user_choice, computer_choice):
    if user_choice == computer_choice:
        return "It's a tie!"
    elif user_choice == "rock":
        if computer_choice == "scissors":
            return "You win!"
        else:
            return "Computer wins!"
    elif user_choice == "paper":
        if computer_choice == "rock":
            return "You win!"
        else:
            return "Computer wins!"
    elif user_choice == "scissors":
        if computer_choice == "paper":
            return "You win!"
        else:
            return "Computer wins!"
    else:
        return "Invalid input!"

def play_game():
    user_choice = get_user_choice()
    computer_choice = get_computer_choice()
    print("You chose:", user_choice)
    print("Computer chose:", computer_choice)
    print(determine_winner(user_choice, computer_choice))

play_game()